### PR TITLE
CRM-19246 acl test for group.get api

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
* [CRM-19246: Add testing for use of ACLs in group ](https://issues.civicrm.org/jira/browse/CRM-19246)

### CODE REVIEW

**Comment by seamuslee001** [comment]
Test this please Jenkins

File: CRM/Contact/BAO/Group.php [hunk 1]
```diff
@@ -900,115 +915,108 @@ public static function getGroupList(&$params) {
     $visibility = CRM_Core_SelectValues::ufVisibility();
 
     while ($object->fetch()) {
-      $permission = CRM_Contact_BAO_Group::checkPermission($object->id, TRUE);
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Group.php]: Per comments I removed the $permission as the code above does already limit the groups based on ACLs & this introduces some caching weirdness when disabled groups are involved. Rest of change is just whitespace.

Tests in CRM_Group_Page_AjaxTest are v thorough

File: CRM/Contact/BAO/Query.php [hunk 2]
```diff
@@ -6322,14 +6322,8 @@ protected function prepareOrderBy($sort, $sortByChar, $sortOrder, $additionalFro
    */
   public function convertGroupIDStringToLabelString(&$dao, $val) {
     $groupIDs = explode(',', $val);
-
-    // The pseudoConstant function does not actually cache.
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: It does seem to now

**Comment by seamuslee001** [comment]
Tested this and works fine to me, Adds some more tests as well looks good

**Comment by eileenmcnaughton** [comment]
Thanks @seamuslee001 - yeah I think it's a good tidy up

**Pull request merged**